report-api/src/test/java/no/sikt/nva/data/report/api/fetch/FetchDataReportTest.java

```diff
@@ -45,29 +62,39 @@ void shouldThrowBadRequest(TestingRequest report)
 
     @ParameterizedTest
     @ArgumentsSource(ValidRequestSource.class)
-    void shouldReturnFormattedResult(TestingRequest request) throws IOException {
-        var service = new QueryService(setupDatabaseConnection());
+    void shouldReturnFormattedResult(TestingRequest request) throws IOException, BadRequestException {
+        var testData = new TestData(List.of(new DatePair(new PublicationDate("2023", "02", "02"),
+                                                     Instant.now().minus(100, ChronoUnit.DAYS)),
+                                        new DatePair(new PublicationDate("2023", "10", "18"),
+                                                     Instant.now().minus(100, ChronoUnit.DAYS))));
+        databaseConnection = setupDatabaseConnection(testData.getModel());
+        var service = new QueryService(databaseConnection);
         var handler = new FetchDataReport(service);
         var input = generateHandlerRequest(request);
         var output = new ByteArrayOutputStream();
         handler.handleRequest(input, output, new FakeContext());
         var response = GatewayResponse.fromOutputStream(output, String.class);
         assertEquals(200, response.getStatusCode());
-        var expected = getExpected(request);
+        var expected = getExpected(request, testData);
         assertEquals(expected, response.getBody());
     }
 
     // TODO: Craft queries and data to test every SELECT clause, BEFORE/AFTER/OFFSET/PAGE_SIZE.
-    private String getExpected(TestingRequest request) {
-        var acceptHeader = request.acceptHeader().get(ACCEPT);
-        var fileExtension = TEXT_CSV.toString().equals(acceptHeader) ? CSV_FILE_EXTENSION : TXT_FILE_EXTENSION;
+    private String getExpected(TestingRequest request, TestData test) throws BadRequestException {
+        var responseType = TEXT_CSV.toString().equals(request.acceptHeader().get(ACCEPT))
+                               ? TEXT_CSV
+                               : TEXT_PLAIN;
+        var data = switch (ReportType.parse(request.reportType())) {
+            case AFFILIATION -> test.getAffiliationResponseData();
+            case CONTRIBUTOR -> test.getContributorResponseData();
+            case FUNDING -> test.getFundingResponseData();
+            case IDENTIFIER -> test.getIdentifierResponseData();
+            case PUBLICATION -> test.getPublicationResponseData();
```

Comment:
The expected result can be calculated from the generated model. The formatting is also generated from the model.